OrbitClientServices/ProcessManager.cpp

```diff
@@ -206,16 +206,16 @@ ErrorMessageOr<std::string> ProcessManagerImpl::LoadProcessMemory(
 }
 
 ErrorMessageOr<std::string> ProcessManagerImpl::LoadNullTerminatedString(
-      int32_t pid, uint64_t address, uint64_t max_size){
-  auto errorOrString = LoadProcessMemory(pid, address, max_size);
-  if(errorOrString.has_value()) {
+    int32_t pid, uint64_t address, uint64_t max_size) {
+  auto error_or_string = LoadProcessMemory(pid, address, max_size);
+  if (error_or_string.has_value()) {
     // Make sure the string is null terminated.
-    std::string str = errorOrString.value();
-    str[str.size()-1] = '\0';
-    errorOrString = str.c_str();
+    std::string str = error_or_string.value();
+    str[str.size() - 1] = '\0';
+    error_or_string = str.c_str();
```

Comment:
Nevermind - it looks like I missed what this line does. I wonder if in the case of there is no \0 this function should return a error? Also from the caller perspective max_size does not make sense and is an implementation detail. We should probably make it a constant on the service side. So that there is no way to request too large chunks of memory from the client.

Reply:
Well if the caller wants to request an unusually long string, having the parameter is handy. I'd prefer keeping it.  I did remove the punching in of the \0 character as it turns out c_str() is guaranteed to be null terminating. Added a comment as well.

Reply:
I still think this should be service-side resposibility. Specifying size for GetNullTerminatedString which returns std::string is leaking implementation details (we shouldn't change api when we change implementation, say implementation reads /proc/pid/mem directly and scans for \0 in place, in which case there is no need for size, it is meaningless). The caller does not have to know how the function works. If you want to get memory with size - you can use existing call, and then do the null-termination logic on the client side.

Reply:
Please see modified code that hides the max_size value as implementation detail and warns if the string isn't null terminated.